src/sentry/processing/realtime_metrics/base.py

```diff
@@ -36,3 +40,78 @@ def increment_project_event_counter(self, project_id: int, timestamp: float) ->
             pipeline.set(key, 0, nx=True, px=self._counter_ttl)
             pipeline.incr(key)
             pipeline.execute()
+
+    # TODO: do these killswitch helpers belong here, or in a different class? LowPriorityQueueStore?
+    # This probably needs locking
+    def get_lpq_projects(self) -> Set[int]:
+        """
+        Fetches the list of projects that are currently using the low priority queue.
+
+        Returns a list of project IDs.
+        """
+        return set(
+            filter(
+                None, {_to_int(project_id) for project_id in self.inner.smembers(LPQ_MEMBERS_KEY)}
+            )
+        )
+
+    def add_project_to_lpq(self, project_id: int) -> None:
+        """
+        Moves projects to the low priority queue.
+
+        This forces all symbolication events triggered by the specified projects to be redirected to
+        the low priority queue, unless these projects have been manually excluded from the low
+        priority queue via the store.symbolicate-event-lpq-never kill switch.
+
+        Raises ``LowPriorityQueueMembershipError`` if this fails to add the specified project to the
+        low priority queue.
+        """
+
+        added = self.inner.sadd(LPQ_MEMBERS_KEY, project_id)
+
+        # Looks like this might already be in the LPQ, or redis failed to add it. This is only a
+        # problem if redis failed to add it.
+        if added == 0 and not self.inner.sismember(LPQ_MEMBERS_KEY, project_id):
+            raise LowPriorityQueueMembershipError(
+                f"Failed to move project ID {project_id} to the low priority queue"
+            )
+
+    def remove_projects_from_lpq(self, project_ids: Set[int]) -> Set[int]:
+        """
+        Removes projects from the low priority queue.
+
+        This restores all specified projects back to the regular queue, unless they have been
+        manually forced into the low priority queue via the store.symbolicate-event-lpq-always kill
+        switch.
+
+
+        Returns all projects that have been successfully removed from the low priority queue.
+        """
+        if len(project_ids) == 0:
+            return set()
```

Comment:
does this special case actually matter?

Reply:
the code below can definitely handle this - it's mostly a preoptimization so we skip talking to redis entirely if for some reason no project IDs are fed in. i'm assuming it's relatively costly to perform that round trip but i could be wrong and this specific check isn't necessary.